### DISCUSSION TITLE
What is the proper way of custom theming?

### DISCUSSION THREAD

**Discussion by JaoHundred** [AvaloniaCommunity/Material.Avalonia#419] [Q&A]
Hi, I'm trying to change a little bit of the default black and white color in the dark and light theme. I searched through the wiki documentation, saw some of discussions posts, and then override the default background brush. It didnt work as expected or I am missing something.

Originally I was doing the app styles and colors in separate files and calling them in the app.axaml and everything worked but the changing of colors, I thought doing it directly in app.axaml could give me a clue but no success, if you need more information please let me know.

![image](https://github.com/user-attachments/assets/666804d1-a916-4dbb-8bd4-468d96db365d)

**Comment by SKProCH** [comment]
Hello, the quick answer is you probably need to declare it like this: 
```
<ResourceDictionary>
    <ResourceDictionary.ThemeDictionaries>
        <ResourceDictionary x:Key='Light'>
            <SolidColorBrush x:Key='BackgroundBrush'>White</SolidColorBrush>
            <SolidColorBrush x:Key='ForegroundBrush'>Black</SolidColorBrush>
        </ResourceDictionary>
        <ResourceDictionary x:Key='Dark'>
            <SolidColorBrush x:Key='BackgroundBrush'>Black</SolidColorBrush>
            <SolidColorBrush x:Key='ForegroundBrush'>White</SolidColorBrush>
        </ResourceDictionary>
    </ResourceDictionary.ThemeDictionaries>
</ResourceDictionary>
```
As described here https://docs.avaloniaui.net/docs/guides/styles-and-resources/resources

Is this help?

**Reply by JaoHundred** [reply]
This is what my colors.axaml looked like. I did some more research and it seems I cant override the default brushes and I'm not sure what is really happening. only the color of the default theme is being applied.

Im doing this right now, the only color being applied is Red, it wont change to any other

App.axml
![image](https://github.com/user-attachments/assets/90c4510d-f92d-4a5e-8361-b3ebac432394)

AppStyle.axaml
![image](https://github.com/user-attachments/assets/08fe8b88-ed43-44c8-a000-db542e880648)

Colors.axaml
![image](https://github.com/user-attachments/assets/51ea967e-cdae-4d40-a82f-a46d05ba6d0d)

![image](https://github.com/user-attachments/assets/30808793-ce2b-42d6-a8cc-05927ec0c203)

**Reply by JaoHundred** [reply]
Did not got it to work, I will leave here my solution to anyone who tried something similar. I dont like my solution but it was the only way for now that I could achieve what I desired.

Name your components that you want to apply a non default color.  After the usual changing theme in material avalonia you can set in code behind or another way like a service class.

My service class is using DI to pick up the view and components that I want to change the default color
![image](https://github.com/user-attachments/assets/14e6d38d-26d3-477e-a0cb-7b542b8f9254)

![image](https://github.com/user-attachments/assets/bb01f6f5-614b-4abb-b99d-10bb36922dbd)
